core/src/test/java/com/radixdlt/rev2/REv2StateComputerTest.java

```diff
@@ -122,15 +121,18 @@ protected void configure() {
 
   private CommittedTransactionsWithProof buildGenesis(LedgerAccumulator accumulator) {
     var initialAccumulatorState = new AccumulatorState(0, HashUtils.zero256());
+    var stake = Decimal.of(1);
     var genesis =
-        TransactionBuilder.createGenesisWithNumValidators(
-            1, Decimal.of(1), UInt64.fromNonNegativeLong(10));
+        TransactionBuilder.createGenesisWithNumValidators(1, stake, UInt64.fromNonNegativeLong(10));
     var accumulatorState =
         accumulator.accumulate(initialAccumulatorState, genesis.getPayloadHash());
     var validatorSet =
         BFTValidatorSet.from(
             PrivateKeys.numeric(1)
-                .map(k -> BFTValidator.from(BFTValidatorId.create(k.getPublicKey()), UInt256.ONE))
```

Comment:
_(note: I am pretty sure that `UInt256.ONE` was a bug here; however, even after fixing it, we cannot add the `check that next_epoch from ledger proof matches the next_epoch returned by RE` logic, because of `null` validator addresses populated by this very line of code)_

Reply:
BTW the same "sloppy mocking" must also be causing `computed state hash at version {} differs from the one in proof` in the `commit()`, which is currently just a `warn!()` - but if we want to have these checks in place one day, we need to rethink the setup of these tests a bit.

Reply:
Capturing my Slack chat with @talekhinezh :

- The validator address should remain an `Option<>`: it is simply up to the StateComputer how it wants to represent validators. Currently the engine always returns the address, but this fact should be opaque to the consensus layer. The current shape of this PR reflects that fine 🟢 
- Some tests (e.g. `test_valid_rev2_transaction_passes()`) currently break this abstraction (and on a much larger scale): they try to mock some StateComputer's outputs, and then pass them to actual StateComputer. Like: call `.commit()` with a LedgerProof that did not come from `.prepare()`. Such tests will soon stop working (when we add some more verifications on receipt hash, validator set, etc). We need to adjust them (as suspected in https://github.com/radixdlt/babylon-node/pull/350#discussion_r1129297454). But this PR still avoids the problem (by not adding these checks just yet) 🟢